packages/basti-cdk/src/basti-access-security-group.ts

```diff
@@ -0,0 +1,65 @@
+import { aws_ec2 } from 'aws-cdk-lib';
+
+import { generateShortId } from './basti-helper';
+import { TARGET_ACCESS_SECURITY_GROUP_NAME_PREFIX } from './basti-constants';
+
+import type { Construct } from 'constructs';
+import type { BastiInstance } from './basti-instance';
+
+/**
+ * The properties for the bastion access security group.
+ */
+export interface BastiAccessSecurityGroupProps
+  extends aws_ec2.SecurityGroupProps {
+  /**
+   * Basti ID
+   *
+   * @default - A 8-character pseudo-random string
+   */
+  readonly bastiId?: string;
+}
+
+/**
+ * The security group for the bastion instance.
+ */
+export class BastiAccessSecurityGroup extends aws_ec2.SecurityGroup {
+  /**
+   * Basti ID
+   */
+  readonly bastiId: string;
```

Comment:
Users might think they have to pass the bastion instance ID here. Let's add a line to the documentation.